File: January 29, 2026.js
function separateLettersAndNumbers(str) {
  
  let lastChar = "";
  let newString = "";

  for (let char of str) {
    if (
      (/[a-zA-Z]/.test(lastChar) && /[0-9]/.test(char)) ||
      (/[0-9]/.test(lastChar) && /[a-zA-Z]/.test(char))
    ) {
      newString += "-";
    }

    newString += char;
    lastChar = char;
  }

  return newString;
}

File: January 29, 2026.py
def separate_letters_and_numbers(s):
    last_char = ""
    new_s = ""

    for char in s:
        if (last_char.isalpha() and char.isdigit()) or (last_char.isdigit() and char.isalpha()):
            new_s += "-"
        
        new_s += char
        last_char = char

    return new_s